libpod/kube.go

```diff
@@ -19,6 +19,40 @@ import (
 	v12 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// GenerateKube generates kubernetes yaml based on a pod or container
+func (r *Runtime) GenerateKube(name string, service bool) (*v1.Pod, *v1.Service, error) {
```

Comment:
Should accept a container/pod struct, or be moved into adapter

Reply:
Why? Input of a name and a bool is not cli oriented.  lots of libpod funcs do the same thing

Reply:
No libpod functions take a name, save for the dedicated container/pod retrieval functions.

Reply:
I'll relent here ... but I did look at a sampling of libpod functions, and they do take names as strings frequently.

Reply:
I saw two, and one of them is actually completely unused (we moved it into pkg/adapter and never removed the original, I think)

Reply:
I like the idea of making this a ctr.GenerateKube function.

Reply:
We'd need a pod generate kube as well, since it also works on pods... You could probably still share almost all of the code (backend function which takes a ctr and a pod and decides what to generate for based on which is not nil, maybe?)

Reply:
I'll see what can be done ...

Reply:
you guys do see that I already have container.GenerateForKube and pod.GenerateForKube right?  The GenerateKube is the generic func that determines which to call. lol

Reply:
Hm. Yeah, that completely eliminates the point of making this run on containers.

I'm back to my original point - GenerateForKube belongs in libpod (runs on a single, specific pod/container and works as advertised), GenerateKube is more of a frontend-y thing (takes user input/name, figures out what to do with it, calls the right backend functions) and probably goes in adapter/, no further refactor of what's here necessary

Reply:
@mheon ptal at whaty I did in the latest push ...